MKdocs Typography, formatting and color improvements

allisonwhilden opened the pull request:
<!-- Thank you for your contribution! Please review https://docs.ag2.ai/docs/contributor-guide/contributing before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?
Cleaning up MKDocs styling as per #1530. This doesn't address everything, but below are details of what it fixes.

------
### Improves styling and information hierarchy for all the headers
I updated the styling for H1 - H5. I kept Jersey10 just for limited usage of H1, but made sure it wasn't uppercase. I switched all headers to be full black instead of grey, adjusted the spacing of all of them, and made H2-H5 Jetbrains Mono.

ex. Before/After
<img width="3277" alt="image" src="https://github.com/user-attachments/assets/b57e4e2e-508f-489f-99d2-1383d6a2f147" />

------
### Improves the colors for dark mode
I tweaked the colors for dark mode to increase the contrast and just look a little more polished.

ex. Before/After
<img width="3277" alt="image" src="https://github.com/user-attachments/assets/18de8830-10dd-4049-9337-341fbfd622f8" />

------
### Cleans up formatting bugs in the user guide section
There were a number of formatting bugs I found while looking through all of the pages of the user guide section. These included:
#### Lack of space around a list making it so the list didn't display like a list
<img width="3277" alt="image" src="https://github.com/user-attachments/assets/86252219-d834-489f-9f79-f8a661f3a0c3" />

#### A H1 used inadvertently elsewhere on the page making it so MKDocs didn't auto-generate an H1 at the top from the front matter
<img width="3277" alt="image" src="https://github.com/user-attachments/assets/14a8a691-1089-44a2-8819-3f5a4d6ec0c6" />

#### A code block that had a code block in it as an output example from an agent which would cause an early end to the code block. I wrapped those code blocks in 4 ticks (````)
<img width="3277" alt="image" src="https://github.com/user-attachments/assets/6522a886-1da8-4fca-bd2a-1dd589c4a796" />

------
### Simplified the layout of the key features on the home page
The centered text of that length is just really ragged and hard to read. Switched to a vertical stack instead of a horizontal layout. The entire home page will be redesigned soon, but this felt like a nice short-term clean-up

Before/after (please note -- the yellow background at the top wasn't removed. I think there was an error with my local server with accessing some of the assets that I haven't solved yet.)

<img width="3277" alt="image" src="https://github.com/user-attachments/assets/bf6c420b-d8b7-47da-a273-ead0a659cd30" />

------
## Related issue number
#1530 (but doesn't close it fully)

## Checks

- [x] I've included any doc changes needed for https://docs.ag2.ai/. See https://docs.ag2.ai/docs/contributor-guide/documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ag2ai/ag2?pullRequest=1566) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ag2ai/ag2?pullRequest=1566) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ag2ai/ag2?pullRequest=1566) it.</sub>

allisonwhilden commented:
@harishmohanraj  I didn't fully remove Jersey10. Instead, I turned off the `uppercase` text transform, darkened the color and increased the font size, and reduced it to only being used on H1. I used Jetbrains Mono on H2-H6. I think this combo reads nicely, while remaining true to our style and brand. Let me know what you think. Thanks!

allisonwhilden commented:
@harishmohanraj @qingyun-wu I have fixed all the checks and added more details above about my changes. Review when you get a chance and let me know what you think and if we can get this ready to merge. Thanks!

davorrunje commented:
Looks great!

harishmohanraj commented:
Hi @allisonwhilden, The font combination looks really good on the website. I have made a couple of minor css changes to your PR.

- Reduced the h1 font size from 3 rem to 2 rem. It looked way too large.
- Fixed a small css fix in use stories page.
